### ISSUE TITLE
Signed parameters that should be unsigned

### ISSUE THREAD

**Issue by Coding-Badly** [smithy-lang/smithy-rs#819]
[This inp parameter](https://awslabs.github.io/aws-sdk-rust/aws_sdk_ec2/client/fluent_builders/struct.RunInstances.html#method.max_count) (and other count parameters) really should be unsigned.  Ideally that particular parameter would be strictly greater than zero but I don't think Rust supports such a thing.  Making count parameters the correct type would introduce the compile-time check (trying to pass a signed value would fail at compile-time).

As far as I can tell Smithy currently does not support unsigned integers so the effort would start there.

**Comment by rcoh** [comment]
Yeah that's correct. Do you have a use case where unsigned numerics are required?
